ArduPlane/precision_landing.cpp

```diff
@@ -0,0 +1,24 @@
+//
+// functions to support precision landing
+//
+
+#include "Plane.h"
+
+#if PRECISION_LANDING == ENABLED
+void Plane::init_precland()
+{
+    plane.precland.init(scheduler.get_loop_rate_hz());
+}
+
+void Plane::update_precland()
+{
+    int32_t height_above_ground_cm = current_loc.alt;
+
+    // use range finder altitude if it is valid, else try to get terrain alt
+    if (rangefinder_alt_ok()) {

```

Comment:
why not use the existing Plane::relative_ground_altitude() function?

Reply:
I'm quite sure we shouldn't do this - and should stick to passing rangefinder data through to the library here.

If we were to drop down to using the terrain database the errors could be really quite large. Better to be safe here and solely rely on rangefinder data IMO - at least for an initial cut.